### PR TITLE
Remove local parameter for get field mapping request

### PR DESCRIPTION
**Pull request by ywangd**
The local parameter of get field mapping request is [marked as deprecated in 7.x](#55099).
This PR removes it for the next major.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/GetFieldMappingsRequest.java [hunk 1]
```diff
@@ -59,26 +57,17 @@ public GetFieldMappingsRequest(StreamInput in) throws IOException {
...
                 throw new IllegalArgumentException("Expected empty type array but received [" + Arrays.toString(types) + "]");
             }
+
         }
         indicesOptions = IndicesOptions.readIndicesOptions(in);
-        local = in.readBoolean();
+        // Consume the deprecated local parameter
+        if (in.getVersion().before(Version.V_8_0_0)) {
+            in.readBoolean();
+        }
```
Comment (ywangd) [inline server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/GetFieldMappingsRequest.java:66]: This is not necessary if we can be sure that the request never goes across wire.

File: server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/GetFieldMappingsRequest.java [hunk 2]
```diff
@@ -133,7 +122,9 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeStringArray(Strings.EMPTY_ARRAY);
         }
         indicesOptions.writeIndicesOptions(out);
-        out.writeBoolean(local);
+        if (out.getVersion().before(Version.V_8_0_0)) {
+            out.writeBoolean(true);
+        }
```
Comment (ywangd) [inline server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/GetFieldMappingsRequest.java:127]: Same here as well, not needed if the request never goes across nodes.

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/2

**Review by martijnvg — APPROVED** [review]
Can you add a note to the `./docs/reference/migration/migrate_8_0/api.asciidoc` file that the `local` param has been removed? LGTM otherwise.

**Comment by ywangd** [comment]
> Can you add a note to the ./docs/reference/migration/migrate_8_0/api.asciidoc file that the local param has been removed? 

Thanks for the reminder. It is [added](https://github.com/elastic/elasticsearch/pull/55100/files#diff-b6a1cb7ef5d9ab019c2b83941a633365R22) now.

**Pull request merged**